### PR TITLE
Time based profile filters

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Allow a profile to be filtered by time of day or days of the week
## Impacts

The captive portal profile filtering
## NEWS file entries

New Features
++++++++++++
- Added the ability to filter a profile by the time of the day
- Added the ability to filter a profile by the days of the week
## Delete branch after merge

NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Please Review

File: lib/pf/profile/filter/day_of_week.pm [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+add a trigger to the value to create update allowed_days
+
+=cut
+
+has '+value' => ( trigger => 1 );
+
+=head1 METHODS
+
+=head2 match
```
Comment (julsemaan) [inline lib/pf/profile/filter/day_of_week.pm:40]: +1 for that documentation header.

This is a perfect example of something that needed additional info and that header covers it

File: lib/pf/profile/filter/day_of_week.pm [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+=head2 _trigger_value
+
+Set allowed_days from the value
+
+=cut
+
+sub _trigger_value {
+    my ($self) = @_;
+    my %allowed_days = map { $_ => undef} split /\s*,\s*/ ,$self->value;
```
Comment (julsemaan) [inline lib/pf/profile/filter/day_of_week.pm]: Could we maybe create that hash with values of 1. Just to make it explicit that it's a truth value.

File: html/pfappserver/lib/pfappserver/I18N/i_default.po [hunk 3]
```diff
@@ -5935,6 +5935,10 @@ msgid "pf::profile::filter::connection_type"
 msgstr "Connection Type"
 
 # pf::filter (Portal Profile Filters)
+msgid "pf::profile::filter::day_of_week"
+msgstr "Day Of Week"
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/I18N/i_default.po]: Any special reason why it shouldn't be written 'Day of week' (without capital letters at the beginning of each word)

File: lib/pf/profile/filter/day_of_week.pm [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    $self->allowed_days(\%allowed_days);
+}
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2014 Inverse inc.
```
Comment (julsemaan) [inline lib/pf/profile/filter/day_of_week.pm]: 2015

File: lib/pf/profile/filter/time_of_day.pm [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    $self->end_time($end);
+}
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2014 Inverse inc.
```
Comment (julsemaan) [inline lib/pf/profile/filter/time_of_day.pm]: 2015

File: lib/pf/profile/filter/time_of_day.pm [hunk 6]
```diff
@@ -0,0 +1,98 @@
+package pf::profile::filter::time_of_day;
+=head1 NAME
+
+pf::profile::filter::time_of_day add documentation
```
Comment (julsemaan) [inline lib/pf/profile/filter/time_of_day.pm]: add documentation ebola virus still happening

File: lib/pf/profile/filter/day_of_week.pm (hunk 1 above)
Comment (extrafu) [inline lib/pf/profile/filter/day_of_week.pm:40]: Comments submitted from Giorgo aren't tolerated here!

File: lib/pf/profile/filter/time_of_day.pm [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+ 
+=head2 _trigger_value
+
+Set start_time and end_time from the value
+
+=cut
+
+sub _trigger_value {
+    my ($self) = @_;
+    my ($start,$end) = split(/-/,$self->value);
```
Comment (julsemaan) [inline lib/pf/profile/filter/time_of_day.pm:65]: This would clearly benefit from validation + error logging.

**Comment by julsemaan** [comment]
You should add validation though pfcmd for these new triggers as they need to follow a specific syntax

File: lib/pf/profile/filter/day_of_week.pm [hunk 8]
```diff
@@ -0,0 +1,101 @@
+package pf::profile::filter::day_of_week;
+=head1 NAME
+
+pf::profile::filter::day_of_week add documentation
```
Comment (julsemaan) [inline lib/pf/profile/filter/day_of_week.pm]: add or remove documentation

File: lib/pf/profile/filter/day_of_week.pm (hunk 1 above)
Comment (julsemaan) [inline lib/pf/profile/filter/day_of_week.pm:40]: Except from the Giorgio in laval. As it welcomes the most jetset customers, it is exempt from any rules governing the lower end society.

**Comment by jrouzierinverse** [comment]
I agree with adding validation. 
However I prefer to do this in a separate pull request since we should also add validation for the other filters.

I would prefer a general solution for all the filters instead just for these two.

**Comment by julsemaan** [comment]
Then I'll suggest we pause this PR. Wait for the validation PR, then adapt this PR to have validation. I find that making an error in these filters is too easy for an end user and having no form of validation will create had to debug cases

**Comment by jrouzierinverse** [comment]
This is going to be replaced by the filter engine

**Pull request closed without merging**